account_move_tax_edition/models/account_move.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import json
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.depends(
+        "line_ids.amount_currency",
+        "line_ids.tax_base_amount",
+        "line_ids.tax_line_id",
+        "partner_id",
+        "currency_id",
+        "amount_total",
+        "amount_untaxed",
+    )
+    def _compute_tax_totals_json(self):
```

Comment:
you shouldn't overwrite function core odoo if not necessary.

May be you can use this

```suggestion
    def _compute_tax_totals_json(self):
        res = super()._compute_tax_totals_json()
        for move in self:
            tax_totals = json.loads(move.tax_totals_json)
            tax_totals["allow_tax_edition"] = (
                move.is_purchase_document(include_receipts=False)
                or move.is_sale_document(include_receipts=False)
            ) and move.state == "draft"
```

it clear and not affect with other module.

Reply:
Thanks, I will improve my code.